interfaces/default/js/lazylibrarian.js

```diff
@@ -0,0 +1,487 @@
+
+    $.ajax({
+        url: WEBDIR + 'lazylibrarian/RefreshAuthor',
+        type: 'post',
+        data: {'authorId': authorId},
+        dataType: 'json',
+        success: function (result) {
+
+        },
+        error: function (req) {
+            console.log('error refreshing author');
+        }
+    })
+}
+
+function searchForAuthor(name, type) {
+    $.ajax({
+        url: WEBDIR + 'lazylibrarian/SearchForAuthor',
+        type: 'get',
+        data: {'name': name,
+                'searchtype': type},
+        dataType: 'json',
+        timeout: 40000,
+        success: function (result) {
+            if (!result || result.length === 0) {
+                $('#add_author_button').attr('disabled', false);
+                return;
+            }
+            // remove any old search
+            $('#add_author_select').html('');
+
+            if (type == 'authorId') {
+                $.each(result, function (index, item) {
+                    var option = $('<option>')
+                    .attr('value', item.authorid)
+                    .attr('author', item.authorname)
+                    .html(item.authorname);
+
+                    $('#add_author_select').append(option);
+                });
+
+            } else {
+                $.each(result, function (index, item) {
+                    var tt;
+                    if (item.bookdate.length) {
+                        // release date should be (yyyy) or empty string
+                        tt = ' (' + item.bookdate.substring(0,4) + ') '
+                    } else {
+                        tt = '  '
+                    }
+                    // item.uniquename == Author name
+                    if (item.authorname === 'None') {
+                        // to remove None..
+                        item.authorname = ''
+                    }
+                    var option = $('<option>')
+                        .attr('value', item.bookid)
+                        .attr('author', item.authorname)
+                        .html(item.bookname + tt + item.authorname);
+
+                    $('#add_author_select').append(option);
+                });
+            }
+            $('#add_author_name').hide();
+            $('#cancel_author_button').show();
+            $('#add_author_select').fadeIn();
+            $('#add_author_button').attr('disabled', false).hide();
+            $('#add_authorid_button').show();
+        }
+    })
+}
+
+function addAuthor(id, searchtype, name, callBack) {
+    // val can be authorId or bookId
+    var stype = (searchtype === 'authorId') ? 'Author' : 'Book';
+    $.ajax({
+        url: WEBDIR + 'lazylibrarian/AddAuthor',
+        data: {'id': id,
+               'searchtype': searchtype},
+        type: 'get',
+        dataType: 'text',
+        success: function (data) {
+            console.log(data);
+            notify('Add ' + stype, 'Successfully added  '+ stype + ' ' + name, 'success');
+            callBack();
```

Comment:
Come to think so it, I was part way rewriting add books last night but forgot! haha